aws/resource_aws_api_gateway_gateway_response.go

```diff
@@ -0,0 +1,152 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayGatewayResponse() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayGatewayResponseCreate,
+		Read:   resourceAwsApiGatewayGatewayResponseRead,
+		Update: resourceAwsApiGatewayGatewayResponseCreate,
+		Delete: resourceAwsApiGatewayGatewayResponseDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rest_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"response_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"status_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"response_templates": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+			},
+
+			"response_parameters": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayGatewayResponseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	log.Printf("[DEBUG] Putting API Gateway Gateway Response")
```

Comment:
Nitpick: I think we can make this log message a lot more useful if we move it down and log the `input` before sending it to the API.

```go
log.Printf("[DEBUG] Putting API Gateway Gateway Response: %s", input)
```